bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/NdkPlugin.kt

```diff
@@ -4,26 +4,34 @@ import com.bugsnag.android.ndk.NativeBridge
 
 internal class NdkPlugin : Plugin {
 
-    companion object {
-        init {
-            System.loadLibrary("bugsnag-ndk")
-        }
-    }
+    private val loader = LibraryLoader()
 
     private external fun enableCrashReporting()
     private external fun disableCrashReporting()
 
     private var nativeBridge: NativeBridge? = null
 
     override fun load(client: Client) {
-        if (nativeBridge == null) {
-            nativeBridge = NativeBridge()
-            client.registerObserver(nativeBridge)
-            client.sendNativeSetupNotification()
-            client.syncInitialState()
+        val loaded = loader.loadLibrary("bugsnag-ndk", client) {
+            val error = it.errors[0]
+            error.errorClass = "NdkLinkError"
+            error.errorMessage = "Native library could not be linked. Bugsnag will not report NDK errors."
```

Comment:
We need to update the Bugsnag message too.

Reply:
I've extracted this message to a constant that is now used in both places.